### DISCUSSION TITLE
Login not working. Only password reset

### DISCUSSION THREAD

**Discussion by hagealex** [Laravel-Backpack/community-forum#529] [General]
Hello,

I'm working on an application with Laravel Backpack for quite some while now. Suddenly, after my last deployment, I can't login into the application dashboard anymore. I always get the error message that the credentials are wrong.

I didn't make any changes on the DB or the code related to auth. I also checked the data in the DB and they seem correct. I was able to reset my password (which means that backpack recognizes the user-email) and was logged in afterwards automaticallly.

Then I logged out and wanted to login with the new PW and...get the same error message.

I have no idea right now what I could do to debug this. In my VM on my local machine I don't get this error. Only on my production environment.

Did someone face a similar issue before or has an idea where to start debugging?

Thanks in advance!

**Comment by karandatwani92** [comment]
can you check using Auth if the previous password works outside the scope of the backpack?

**Reply by hagealex** [reply]
I tried the same PW on my local VM and it works. Both systems (local VM and prod) use the same backpack version.

**Comment by o15a3d4l11s2** [comment]
You can use my suggestion from another similar thread: https://github.com/Laravel-Backpack/community-forum/discussions/357#discussioncomment-4654325. For completeness I am also copying it here.

If you are using the default settings for hashing (Bcrypt), you could try changing one password directly in the database to `$2a$10$LLHgjgY2SNlrIELWFG1Q6O4hlHG42WllWNjISZXUNfKGlLeSKAeOy`, which is equivalent to `12345678` and then try to login with this password (`12345678`). If it succeeds, it means the problem is related to the hash generation. If it fails, it means it is related to the login part that checks if the password is correct.

After that please check https://github.com/Laravel-Backpack/community-forum/discussions/357#discussioncomment-4677362 which will then test if hashing itself works properly.

**Reply by hagealex** [reply]
Thanks for your suggestion!

I tried it the following: I reset my PW and used `12345678`. It was stored in the DB as `$2y$10$wCS3Q3LnkJQ7SE1FrwdLl.BWipITeuSMwyGQkBibhjDK.RX1jBY/6`. This differs from what you expected. I checked my Laravel config and the hashing driver is `bcrypt`. I didn't change that.

Weird: Now the login is working again. I also changed the PW to the old one and this also works.

I have absolutely no idea what's going on here :sweat_smile:

**Reply by o15a3d4l11s2** [reply]
If you hash the same password 10 times, you will get 10 different hash results. So it is normal that the hashes differ.

**Reply by hagealex** [reply]
Ah, okay.

Well as I can login again now without having changed anything on the code or my settings I hope this issue was just some weird random stuff that occurred for...reasons.

Thanks for your support here!
